src/main/java/org/commcare/formplayer/api/json/JsonActionUtils.java

```diff
@@ -113,7 +115,12 @@ private static JSONObject questionAnswerToJson(FormEntryController controller,
                 return ret;
             }
         }
-        int result = controller.answerQuestion(prompt.getIndex(), answerData);
+	IAnswerData currentAnswerData = prompt.getAnswerValue();
```

Comment:
One quick note: I think this might compute more information than intended, and that it might be equally valid to retrieve the value directly from the model (IE: Model->getform->getelement->getvalue with the index)